app/javascript/components/Results/SpellingSuggestion/SpellingSuggestion.tsx

```diff
@@ -1,18 +1,68 @@
 /* eslint-disable camelcase */
-import React, { useContext } from 'react';
+import React, { useContext, useEffect } from 'react';
 import { GridContainer, Grid } from '@trussworks/react-uswds';
 import { LanguageContext } from '../../../contexts/LanguageContext';
 import parse from 'html-react-parser';
-
+import { clickTracking } from '../../../utils';
+import { moduleCode } from '../../../utils/constants';
 import './SpellingSuggestion.css';
-
 interface SpellingSuggestionProps {
   suggested: string;
   original: string;
+  originalUrl: string;
+  originalQuery: string;
+  suggestedQuery: string;
+  suggestedUrl: string;
+  affiliate: string;
+  vertical: string;
 }
 
-export const SpellingSuggestion = ({ suggested, original }: SpellingSuggestionProps) => {
+export const SpellingSuggestion = ({ suggested, original, originalQuery, originalUrl, suggestedUrl, suggestedQuery, affiliate, vertical }: SpellingSuggestionProps) => {
   const i18n = useContext(LanguageContext);
+  const getUrl = (url: string) => window.location.origin + url;
+  const position = 1;
+
+  const clickTrackingModuleMap = {
+    web: {
+      suggestedQueryModule: moduleCode.spellingSuggestionsBing,
+      originalQueryModule: moduleCode.spellingOverridesBing
+    },
+    blended: {
+      suggestedQueryModule: moduleCode.spellingSuggestionsSearch
+    },
+    i14y: {
+      suggestedQueryModule: moduleCode.spellingSuggestionsSearch,
+      originalQueryModule: moduleCode.spellingOverridesI14
+    },
+    image: {
+      suggestedQueryModule: moduleCode.spellingSuggestionsImages
+    }
+    // UNABLE TO FIND THE MODULE CODE
```

Comment:
I *believe* the 5 verticals that are possible here are:

* blended
* docs
* i14y
* image
* news
* web

I believe the ones you're missing - docs and news - should just get the generic search codes as a baseline.  (`spellingSuggestionsSearch` and `spellingOverridesSearch`)

I don't know if it simplifies things to set a default starting value for these codes as the generic ones above, and then only specifically changes those codes in the following cases:
* if vertical == 'web' change both the override and suggestion code to the bing codes
* if vertical == 'i14y' change only the override to the i14y override code
* if vertical == 'images' change only the suggestion to the images suggestion code

Also, nitpicky, but as we work through this let's making sure we're naming things `i14y` and not `i14` so we can keep consistent throughout the codebase.

Reply:
Thank you @lorawoodford for your feedback. I will check if default values can be used.

Reply:
Hi @lorawoodford - code is updated to address the above mentioned suggestions.
https://github.com/GSA/search-gov/pull/1507/files#diff-eb8002783517dbc585bf3ce80e9941905e8403e6087b213abf95134a5dad18eaR25